substrate/runtime/src/lib.rs

```diff
@@ -304,6 +361,63 @@ impl validator_sets::Config for Runtime {
   type RuntimeEvent = RuntimeEvent;
 }
 
+pub struct AssetConverter;
+impl MultiAssetIdConverter<Coin, Coin> for AssetConverter {
+  /// Returns the MultiAssetId representing the native currency of the chain.
+  fn get_native() -> Coin {
+    Coin::Serai
+  }
+
+  /// Returns true if the given MultiAssetId is the native currency.
+  fn is_native(asset: &Coin) -> bool {
+    *asset == Coin::Serai
+  }
+
+  /// If it's not native, returns the AssetId for the given MultiAssetId.
+  fn try_convert(asset: &Coin) -> Result<Coin, ()> {
+    if *asset == Coin::Serai {
+      return Err(());
+    }
+    Ok(*asset)
+  }
+
+  /// Wraps an AssetId as a MultiAssetId.
+  fn into_multiasset_id(asset: &Coin) -> Coin {
+    *asset
+  }
+}
+
+impl dex::Config for Runtime {
+  type RuntimeEvent = RuntimeEvent;
+  type Currency = Balances;
+  type Balance = SubstrateAmount;
+  type AssetBalance = SubstrateAmount;
+  type HigherPrecisionBalance = u128;
+
+  type AssetId = Coin;
+  type MultiAssetId = Coin;
+  type MultiAssetIdConverter = AssetConverter;
+  type PoolAssetId = u32;
```

Comment:
That is what I tried initially but lp tokens are constructed on the fly. So pallet should be able to auto make ids for them. Hence, `PoolAssetId` should be something that is incremental, something that implements `Incrementable`. 

I guess we can still implement this for an enum? But enum will only have certain elements and this will create a new lp token per pool automatically.  I mean we will only have certain amount of pools right? so enum still makes sense? yea I was confused there too.

Reply:
And then we can't guarantee the increment matches the intended order of Coin.

Got it. This is fine for now :+1: